### PR TITLE
add delete file functionality to TN_S3_file_uploader

### PR DESCRIPTION
**Pull request by sammarz**
https://thinkneardev.atlassian.net/browse/REX-2613

### CODE REVIEW

File: lib/tn_s3_file_uploader/cli_parser.rb [hunk 1]
```diff
@@ -24,6 +24,12 @@ def parse_cmd_line(args)
           options[:s3_output_pattern] = s3_output_pattern
         end
 
+        options[:delete_log_files_flag] = true
+        opts.on("-d", "--delete-log-files [DELETE_LOG_FILES]", "The flag for deleting log files"\
+            "It is an optional parameter, default is true, decides if log files are deleted after successful upload") do |delete_log_files_flag|
```
Comment (eduardog) [inline lib/tn_s3_file_uploader/cli_parser.rb]: This is used by both our SymmetricDS and venice instances.  We're currently only changing venice' configs.
Even though we could choose to not upgrade without handling this, I'd argue the default should match previous behavior (not deleting logs).

File: lib/tn_s3_file_uploader/cli_parser.rb [hunk 2]
```diff
@@ -24,6 +24,12 @@ def parse_cmd_line(args)
           options[:s3_output_pattern] = s3_output_pattern
         end
 
+        options[:delete_log_files_flag] = false
+        opts.on("-d", "--delete-log-files [DELETE_LOG_FILES]", "The flag for deleting log files"\
+            "It is an optional parameter, default is true, decides if log files are deleted after successful upload") do |delete_log_files_flag|
+          options[:s3_output_pattern] = delete_log_files_flag
```
Comment (anastasiosl) [inline lib/tn_s3_file_uploader/cli_parser.rb]: We should set the delete_log_files_flag to ```options[:delete_log_files]```, not override the ```options[:s3_output_pattern]```

**Comment by anastasiosl** [comment]
I want to see a new spec in ```cli_parser_spec``` that checks we set ```delete-log-files``` to true by default.

**Comment by anastasiosl** [comment]
Rest looks good.

Done with first pass.

File: lib/tn_s3_file_uploader/cli_parser.rb [hunk 3]
```diff
@@ -24,6 +24,12 @@ def parse_cmd_line(args)
           options[:s3_output_pattern] = s3_output_pattern
         end
 
+        options[:delete_log_files_flag] = false
```
Comment (anastasiosl) [inline lib/tn_s3_file_uploader/cli_parser.rb]: Default should be true, right?

Comment (sammarz) [inline lib/tn_s3_file_uploader/cli_parser.rb]: we decided it to be false

**Comment by anastasiosl** [comment]
Looks good.

**Pull request merged**